Plugin stops working because of Side effect

### Before submitting your bug report

- [ ] I've tried using the "Ask AI" feature on the [Continue docs site](https://docs.continue.dev/) to see if the docs have an answer
- [x] I'm not able to find a related conversation on [GitHub discussions](https://github.com/continuedev/continue/discussions) that reports the same bug
- [x] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue) that reports the same bug
- [x] I've seen the [troubleshooting guide](https://docs.continue.dev/troubleshooting) on the Continue Docs

### Relevant environment info

```Markdown
- OS: Win11 24H2
- Continue version: 1.0.55
- IDE version: Webstorm 2025.3.1
- Model: Codestral
```

### Description

From time to time the Plugin stops working and is no longer reacting
I am getting the following error message:

java.lang.AssertionError: Side effect not allowed: INVOKE_LATER
	at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:91)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:439)
	at com.intellij.util.EventDispatcher.throwExceptions(EventDispatcher.java:170)
	at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:128)
	at com.intellij.util.EventDispatcher.lambda$doCreateMulticaster$1(EventDispatcher.java:85)
	at jdk.proxy2/jdk.proxy2.$Proxy92.documentChanged(Unknown Source)
	at com.intellij.openapi.editor.impl.DocumentImpl.lambda$changedUpdate$6(DocumentImpl.java:974)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeNonCancelableSection$3(CoreProgressManager.java:325)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:741)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:333)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:144)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:333)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:324)
	at com.intellij.openapi.editor.impl.DocumentImpl.changedUpdate(DocumentImpl.java:971)
	at com.intellij.openapi.editor.impl.DocumentImpl.updateText(DocumentImpl.java:873)
	at com.intellij.openapi.editor.impl.DocumentImpl.replaceString(DocumentImpl.java:709)
	at com.intellij.openapi.editor.impl.DocumentImpl.replaceString(DocumentImpl.java:640)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.doCommitTransaction(PsiToDocumentSynchronizer.java:235)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.lambda$commitTransaction$1(PsiToDocumentSynchronizer.java:211)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.lambda$doSync$0(PsiToDocumentSynchronizer.java:107)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.performAtomically(PsiToDocumentSynchronizer.java:125)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.doSync(PsiToDocumentSynchronizer.java:107)
	at com.intellij.psi.impl.PsiToDocumentSynchronizer.commitTransaction(PsiToDocumentSynchronizer.java:211)
	at com.intellij.pom.core.impl.PomModelImpl.commitTransaction(PomModelImpl.java:193)
	at com.intellij.pom.core.impl.PomModelImpl.lambda$runTransaction$1(PomModelImpl.java:149)
	at com.intellij.psi.impl.DebugUtil.performPsiModification(DebugUtil.java:570)
	at com.intellij.pom.core.impl.PomModelImpl.lambda$runTransaction$2(PomModelImpl.java:101)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeNonCancelableSection$3(CoreProgressManager.java:325)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:792)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:748)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:333)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:150)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:333)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:324)
	at com.intellij.pom.core.impl.PomModelImpl.runTransaction(PomModelImpl.java:90)
	at com.intellij.psi.impl.source.tree.ChangeUtil.prepareAndRunChangeAction(ChangeUtil.java:145)
	at com.intellij.psi.impl.source.tree.CompositeElement.replaceChild(CompositeElement.java:654)
	at com.intellij.lang.javascript.psi.impl.JSElementImpl.replaceInParent(JSElementImpl.java:162)
	at com.intellij.lang.javascript.psi.impl.JSStubElementImpl.replace(JSStubElementImpl.java:196)
	at com.intellij.lang.javascript.inspections.ES6ShorthandObjectPropertyInspection.replacePropertyWithText(ES6ShorthandObjectPropertyInspection.java:297)
	at com.intellij.lang.javascript.inspections.ES6ShorthandObjectPropertyInspection$ExpandFunctionPropertyQuickFix.lambda$invoke$0(ES6ShorthandObjectPropertyInspection.java:274)
	at com.intellij.lang.javascript.inspections.ES6ShorthandObjectPropertyInspection.transformProperty(ES6ShorthandObjectPropertyInspection.java:285)
	at com.intellij.lang.javascript.inspections.ES6ShorthandObjectPropertyInspection$ExpandFunctionPropertyQuickFix.invoke(ES6ShorthandObjectPropertyInspection.java:272)
	at com.intellij.codeInspection.LocalQuickFixOnPsiElement.applyFix(LocalQuickFixOnPsiElement.java:97)
	at com.intellij.codeInspection.LocalQuickFixOnPsiElement.applyFix(LocalQuickFixOnPsiElement.java:87)
	at com.intellij.codeInspection.LocalQuickFixOnPsiElement.applyFix(LocalQuickFixOnPsiElement.java:22)
	at com.intellij.codeInspection.LocalQuickFix.generatePreview(LocalQuickFix.java:86)
	at com.intellij.codeInspection.ex.QuickFixWrapper.generatePreview(QuickFixWrapper.java:237)
	at com.intellij.codeInsight.intention.impl.IntentionActionWithTextCaching$MyIntentionAction.generatePreview(IntentionActionWithTextCaching.java:264)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview$lambda$0$0$0(IntentionPreviewComputable.kt:133)
	at com.intellij.model.SideEffectGuard$Companion.computeWithAllowedSideEffectsBlocking$lambda$0(SideEffectGuard.kt:42)
	at com.intellij.concurrency.ThreadContext.installThreadContext(threadContext.kt:305)
	at com.intellij.model.SideEffectGuard$Companion.computeWithAllowedSideEffectsBlocking(SideEffectGuard.kt:41)
	at com.intellij.model.SideEffectGuard$Companion.computeWithoutSideEffects(SideEffectGuard.kt:27)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview$lambda$0$0(IntentionPreviewComputable.kt:132)
	at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.lambda$postponeFormattingInside$2(PostprocessReformattingAspectImpl.java:137)
	at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.postponeFormattingInside(PostprocessReformattingAspectImpl.java:146)
	at com.intellij.psi.impl.source.PostprocessReformattingAspectImpl.postponeFormattingInside(PostprocessReformattingAspectImpl.java:136)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview$lambda$0(IntentionPreviewComputable.kt:131)
	at com.intellij.codeInsight.intention.preview.IntentionPreviewUtils.previewSession(IntentionPreviewUtils.java:132)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.invokePreview(IntentionPreviewComputable.kt:130)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.generatePreview(IntentionPreviewComputable.kt:99)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.tryCreateDiffContent(IntentionPreviewComputable.kt:75)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewComputable.call(IntentionPreviewComputable.kt:58)
	at com.intellij.codeInsight.intention.impl.IntentionHintComponent$IntentionPopup.lambda$getPreviewHandler$4(IntentionHintComponent.java:869)
	at com.intellij.codeInsight.intention.impl.preview.PreviewHandler.lambda$new$0(PreviewHandler.java:57)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewPopupUpdateProcessor$updatePopup$5.invokeSuspend$lambda$0(IntentionPreviewPopupUpdateProcessor.kt:119)
	at com.intellij.openapi.application.rw.InternalReadAction.insideReadAction(InternalReadAction.kt:111)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable$lambda$0(InternalReadAction.kt:101)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$0$0$0(cancellableReadAction.kt:32)
	at com.intellij.platform.locking.impl.NestedLocksThreadingSupport.tryRunReadAction(NestedLocksThreadingSupport.kt:902)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1237)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$0$0(cancellableReadAction.kt:30)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:73)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:152)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$0(cancellableReadAction.kt:28)
	at com.intellij.concurrency.ThreadContext.installThreadContext(threadContext.kt:305)
	at com.intellij.concurrency.ThreadContext.installThreadContext$default(threadContext.kt:303)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal(cancellableReadAction.kt:24)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable(InternalReadAction.kt:100)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadAction(InternalReadAction.kt:86)
	at com.intellij.openapi.application.rw.InternalReadAction.readLoop(InternalReadAction.kt:73)
	at com.intellij.openapi.application.rw.InternalReadAction.access$readLoop(InternalReadAction.kt:15)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$6.invokeSuspend(InternalReadAction.kt:52)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$6.invoke(InternalReadAction.kt)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$6.invoke(InternalReadAction.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:44)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:166)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.openapi.application.rw.InternalReadAction.runReadAction(InternalReadAction.kt:50)
	at com.intellij.openapi.application.rw.PlatformReadWriteActionSupport.executeReadAction(PlatformReadWriteActionSupport.kt:66)
	at com.intellij.openapi.application.ReadWriteActionSupport.executeReadAction$default(ReadWriteActionSupport.kt:15)
	at com.intellij.openapi.application.CoroutinesKt.constrainedReadAction(coroutines.kt:60)
	at com.intellij.openapi.application.CoroutinesKt.readAction(coroutines.kt:27)
	at com.intellij.codeInsight.intention.impl.preview.IntentionPreviewPopupUpdateProcessor$updatePopup$5.invokeSuspend(IntentionPreviewPopupUpdateProcessor.kt:118)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:610)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runDefaultDispatcherTask(CoroutineScheduler.kt:1194)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:906)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:775)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:762)
Caused by: java.lang.Throwable: Control-flow exceptions (e.g. this class com.intellij.model.SideEffectGuard$SideEffectNotAllowedException) should never be logged. Instead, these should have been rethrown if caught.
	at com.intellij.openapi.diagnostic.Logger.ensureNotControlFlow(Logger.java:504)
	at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:84)
	... 101 more
Caused by: com.intellij.model.SideEffectGuard$SideEffectNotAllowedException: Side effect not allowed: INVOKE_LATER
	at com.intellij.model.SideEffectGuard$Companion.checkSideEffectAllowed(SideEffectGuard.kt:21)
	at com.intellij.model.SideEffectGuard.checkSideEffectAllowed(SideEffectGuard.kt)
	at com.intellij.openapi.application.impl.LaterInvocator.invokeLater(LaterInvocator.java:98)
	at com.intellij.openapi.application.impl.ApplicationImpl.dispatchCoroutineOnEDT(ApplicationImpl.java:433)
	at com.intellij.openapi.application.impl.EdtCoroutineDispatcher.dispatch(EdtCoroutineDispatcher.kt:48)
	at kotlinx.coroutines.internal.DispatchedContinuationKt.safeDispatch(DispatchedContinuation.kt:254)
	at kotlinx.coroutines.internal.DispatchedContinuationKt.resumeCancellableWith(DispatchedContinuation.kt:318)
	at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:26)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:358)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:134)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:52)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:43)
	at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source)
	at com.github.continuedev.continueintellijextension.listeners.DocumentChangeTracker.handleDocumentChange(DocumentChangeTracker.kt:45)
	at com.github.continuedev.continueintellijextension.listeners.DocumentChangeTracker.documentChanged(DocumentChangeTracker.kt:24)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:120)
	... 98 more
	Suppressed: com.intellij.util.ExceptionUtilRt$RethrownStack: Rethrown at
		at com.intellij.util.ExceptionUtilRt.addRethrownStackAsSuppressed(ExceptionUtilRt.java:43)
		at com.intellij.util.ExceptionUtilRt.rethrowUnchecked(ExceptionUtilRt.java:17)
		at com.intellij.util.ExceptionUtil.rethrowUnchecked(ExceptionUtil.java:131)
		at com.intellij.util.ExceptionUtil.rethrow(ExceptionUtil.java:146)
		at com.intellij.idea.IdeaLogger.error(IdeaLogger.java:191)
		at com.intellij.openapi.diagnostic.Logger.error(Logger.java:426)
		at com.intellij.openapi.application.impl.ExceptionsKt.logExceptionSafely(exceptions.kt:58)
		at com.intellij.openapi.application.impl.ExceptionsKt.processUnhandledException(exceptions.kt:51)
		at com.intellij.openapi.application.impl.CoroutineExceptionHandlerImpl.handleException(CoroutineExceptionHandlerImpl.kt:27)
		at kotlinx.coroutines.internal.CoroutineExceptionHandlerImpl_commonKt.handleUncaughtCoroutineException(CoroutineExceptionHandlerImpl.common.kt:34)
		at kotlinx.coroutines.CoroutineExceptionHandlerKt.handleCoroutineException(CoroutineExceptionHandler.kt:31)
		at kotlinx.coroutines.StandaloneCoroutine.handleJobException(Builders.common.kt:194)
		at kotlinx.coroutines.JobSupport.finalizeFinishingState(JobSupport.kt:222)
		at kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath(JobSupport.kt:946)
		at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:894)
		at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:859)
		at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:99)
		at kotlinx.coroutines.intrinsics.CancellableKt.dispatcherFailure(Cancellable.kt:62)
		at kotlinx.coroutines.intrinsics.CancellableKt.startCoroutineCancellable(Cancellable.kt:75)
		... 109 more
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelled}@4c5b5582, Dispatchers.Main]


### To reproduce

_No response_

### Log output

```Shell

```

--- a/extensions/intellij/src/main/kotlin/com/github/continuedev/continueintellijextension/listeners/ActiveHandlerManager.kt
+++ b/extensions/intellij/src/main/kotlin/com/github/continuedev/continueintellijextension/listeners/ActiveHandlerManager.kt
@@ -56,7 +56,7 @@ interface CursorMovementHandler {
 @Service(Service.Level.PROJECT)
 class ActiveHandlerManager(private val project: Project) : SelectionListener, CaretListener, DumbAware {
 
-    private val coroutineScope = CoroutineScope(Dispatchers.Default + SupervisorJob())
+    private val coroutineScope = CoroutineScope(Dispatchers.Default.limitedParallelism(1) + SupervisorJob())
     private var activeHandler: CursorMovementHandler? = null
     private var isHandlingEvent = false
 

--- a/extensions/intellij/src/main/kotlin/com/github/continuedev/continueintellijextension/listeners/DocumentChangeTracker.kt
+++ b/extensions/intellij/src/main/kotlin/com/github/continuedev/continueintellijextension/listeners/DocumentChangeTracker.kt
@@ -13,7 +13,7 @@ import kotlinx.coroutines.*
 @Service(Service.Level.PROJECT)
 class DocumentChangeTracker(private val project: Project) : DocumentListener {
 
-    private val coroutineScope = CoroutineScope(Dispatchers.Default + SupervisorJob())
+    private val coroutineScope = CoroutineScope(Dispatchers.Default.limitedParallelism(1) + SupervisorJob())
     private var typingHandler: TypingSessionHandler? = null
     private var typingSessionTimer: Job? = null
 

